src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -206,11 +201,11 @@ public static unsafe void RhUnboxAny(object? o, ref byte data, MethodTable* pUnb
         [RuntimeExport("RhUnbox2")]
         public static unsafe ref byte RhUnbox2(MethodTable* pUnboxToEEType, object obj)
         {
-            if ((obj == null) || !UnboxAnyTypeCompare(obj.GetMethodTable(), pUnboxToEEType))
+            if (!UnboxAnyTypeCompare(obj.GetMethodTable(), pUnboxToEEType))
```

Comment:
```suggestion
            // This will throw NullReferenceException if obj is null.
            if (!UnboxAnyTypeCompare(obj.GetMethodTable(), pUnboxToEEType))
```